zproject/backends.py

```diff
@@ -1423,7 +1423,21 @@ def auth_url(self) -> str:
         relay_state = self.put_data_in_redis(data_to_relay)
 
         return auth.login(return_to=relay_state)
-
+    
+    def _check_entitlements(self, idp, attributes):
+        """
+        Check if we require the presence of any specific eduPersonEntitlement.
+        raise AuthForbidden if the user should not be authenticated, or do nothing
+        to allow the login pipeline to continue.
+        """
+        if "requiredEntitlements" in idp.conf:
+            entitlements = attributes.get(OID_EDU_PERSON_ENTITLEMENT, [])
```

Comment:
Why is `OID_EDU_PERSON_ENTITLEMENT`  the right attribute to get?  Is that only for EDU, and there's some other entitlements object one needs to check as well?

I don't see that field documented in https://python-social-auth.readthedocs.io/en/latest/backends/saml.html, and don't know a lot about SAML entitlements, so any resources would be appreciated.

Reply:
@timabbott hi, eduPersonEntitlement is an attribute for users. This attribute holds person's all affiliates. this is an enum: `OID_EDU_PERSON_ENTITLEMENT` which is provided by python saml. 
There are others like below:
```
# constants:
OID_COMMON_NAME = "urn:oid:2.5.4.3"
OID_EDU_PERSON_PRINCIPAL_NAME = "urn:oid:1.3.6.1.4.1.5923.1.1.1.6"
OID_EDU_PERSON_ENTITLEMENT = "urn:oid:1.3.6.1.4.1.5923.1.1.1.7"
OID_GIVEN_NAME = "urn:oid:2.5.4.42"
OID_MAIL = "urn:oid:0.9.2342.19200300.100.1.3"
OID_SURNAME = "urn:oid:2.5.4.4"
OID_USERID = "urn:oid:0.9.2342.19200300.100.1.1"
```
as you see it is a standart representation for the person entitlements.
documentation is too weak for python saml, it did get info a lot of researching and code analyzing...

only info for _check_entitlements is at the end of the page: https://python-social-auth.readthedocs.io/en/latest/backends/saml.html